shared/libraries/config_protocol/include/config_protocol/config_protocol_server.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2022-2023 Blueberry d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <config_protocol/config_protocol.h>
+#include <opendaq/device_ptr.h>
+
+namespace daq::config_protocol
+{
+
+class ConfigProtocolServer
+{
+public:
+    ConfigProtocolServer(DevicePtr rootDevice);
+
+    PacketBuffer addPacket(const PacketBuffer& packetBuffer);
+    PacketBuffer addPacket(void *mem);
+    PacketBuffer addPacket(void* mem, DeleterCallback deleterCallback);
+
+private:
+    DevicePtr rootDevice;
+    DeserializerPtr deserializer;
+    SerializerPtr serializer;
+    std::unordered_map<std::string, std::function<BaseObjectPtr(const DictPtr<IString, IBaseObject>&)>> rpcDispatch;
+
+    PacketBuffer processPacket(const PacketBuffer& packetBuffer);
+    StringPtr processRpc(const StringPtr& jsonStr);
+
+    BaseObjectPtr callRpc(const StringPtr& name, const DictPtr<IString, IBaseObject>& params);
+
+    BaseObjectPtr getComponent(const DictPtr<IString, IBaseObject>& params) const;
+    static ComponentPtr findComponent(const ComponentPtr& component, const std::string& id);
```

Comment:
There should probably be an internal hashmap of all components available either within the server, or somewhere that's accessible by all modules. Maybe in the context?

Reply:
Perhaps, but at the moment I do not expect performance issues and hashmap needs to be updated when new components are added and removed.